### PR TITLE
Topic recognition - Zachary Oar (45314669) - ISICs data set with the Improved UNet

### PR DESCRIPTION
**Pull request by zac-oar-uq**
### Author
Zachary Oar (45314669)

### Problem
Segment the ISICs data set with the Improved UNet with all labels having a minimum Dice similarity coefficient of 0.8 on the test set.

### Included Files
Includes a file for the model of the Improved UNet and a driver script to run training/testing on the ISIC data set.  Also includes 3 png images and a README file for documentation.

`model.py` is the model file and `driver.py` is the driver script.  The driver script can be ran using an IDE or with the command line, assuming all dependencies have been installed as per the README.

### Description
The `model.py` file includes the function `make_model()`, which will return a convolutional neural network model with the Improved UNet structure using Keras.

The `driver.py` script can be ran on the command line with:
```
>python driver.py
```
Running this script will prepare the ISIC2018 Challenge images for training/testing, train the model from `model.py` over 15 epochs, evaluate the trained model on a test set of images and plot a set of predicted images from the test set.

For more information, please see the README.

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Easy Difficulty
- Did not report Quantitative results 

Other Feedback
I would put the code you actually run in `if __name__ == '__main__'`

**Comment by zac-oar-uq** [comment]
> * Sufficiently detailed commit messages
> * Code valid for Normal Difficulty
> * Did not report Quantitative results
> 
> Other Feedback
> I would put the code you actually run in `if __name__ == '__main__'`

I have now added some reporting of quantitative results to the README and put the running code into the main block.

**Comment by rexsw** [comment]
Code looks good for normal difficulty

This is just some feedback you don’t have to make these changes

Tensorflow has functions for one hot encoding there’s no need to write your own. This also means you don’t need to use libraries such as numpy in your project. The same thing applies to sklearn for train_test split and PIL for image loading. Tensorflow can do these things so there’s no need to have all these libraries be a part of your project.  

Comments such as 
   # evaluate the model on the test set
    model.evaluate(test_gen)
Are not helpful for the readability of the code and should be left out. But overall well commented

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. remove unnecessary .gitignore file that is causing conflict.**

**Comment by zac-oar-uq** [comment]
> **Feedback required. remove unnecessary .gitignore file that is causing conflict.**

I have removed the .gitignore file conflicts and reverted it back to what it was originally.
